src/install/yarn.zig

```diff
@@ -0,0 +1,1719 @@
+                        .prod = dep_type == .production,
+                        .optional = dep_type == .optional,
+                        .dev = dep_type == .development,
+                        .peer = dep_type == .peer,
+                        .workspace = dep_entry.workspace,
+                    },
+                };
+                var found_package_id: ?Install.PackageID = null;
+                outer: for (yarn_lock_.entries.items, 0..) |entry_, yarn_idx| {
+                    for (entry_.specs) |entry_spec| {
+                        if (strings.eql(entry_spec, dep_spec)) {
+                            found_package_id = yarn_entry_to_package_id[yarn_idx];
+                            break :outer;
+                        }
+                    }
+                }
+
+                if (found_package_id) |pkg_id| {
+                    res_buf[count] = pkg_id;
+                    count += 1;
+                }
+            }
+        }
+        return res_buf[0..count];
+    }
+}.process;
+
+pub fn migrateYarnLockfile(
+    this: *Lockfile,
+    manager: *Install.PackageManager,
+    allocator: Allocator,
+    log: *logger.Log,
+    data: string,
+    abs_path: string,
+) !LoadResult {
+    var yarn_lock = YarnLock.init(allocator);
+    defer yarn_lock.deinit();
+
+    try yarn_lock.parse(data);
+
+    this.initEmpty(allocator);
+    Install.initializeStore();
+
+    var string_buf = this.stringBuf();
+
+    // read package.json to get specified dependencies
+    var path_buf: bun.PathBuffer = undefined;
+    const package_json_dir = std.fs.path.dirname(abs_path) orelse ".";
+    const package_json_path = bun.path.joinAbsStringBufZ(package_json_dir, &path_buf, &.{"package.json"}, .auto);
+
+    var num_deps: u32 = 0;
+    var root_dep_count: u32 = 0;
+    var root_dep_count_from_package_json: u32 = 0;
+
+    var root_dependencies = std.ArrayList(struct { name: []const u8, version: []const u8, dep_type: DependencyType }).init(allocator);
+    defer {
+        for (root_dependencies.items) |dep| {
+            allocator.free(dep.name);
+            allocator.free(dep.version);
+        }
+        root_dependencies.deinit();
+    }
+
+    const package_json_contents = bun.sys.File.readFrom(bun.FD.cwd(), package_json_path, allocator).unwrap() catch |err| {
+        Output.errGeneric("Failed to read package.json: {s}", .{@errorName(err)});
+        Global.exit(1);
+    };
+    defer allocator.free(package_json_contents);
+
+    const package_json_source = logger.Source.initPathString(package_json_path, package_json_contents);
+    const package_json_expr = JSON.parsePackageJSONUTF8WithOpts(
+        &package_json_source,
+        log,
+        allocator,
+        .{
+            .is_json = true,
+            .allow_comments = true,
+            .allow_trailing_commas = true,
+            .guess_indentation = true,
+        },
+    ) catch |err| {
+        Output.errGeneric("Failed to parse package.json: {s}", .{@errorName(err)});
```

Comment:
Let's avoid crashing and instead let's just silently ignore

return error.InvalidPackageJSON